File: src/main/java/service/ReservationService.java
package service;

import domain.Reservation;
import domain.Trip;
import repository.ReservationRepo;
import utils.Pair;

import java.util.Collection;

public class ReservationService {
    private ReservationRepo repo;

    public ReservationService(ReservationRepo repo) {
        this.repo = repo;
    }

    public void saveReservation(String client, Trip trip, String phoneNumber, int seats) {
        repo.save(new Reservation(new Pair<>(client, trip), phoneNumber, seats));
    }

    public Collection<Reservation> getAllReservations() { return repo.getAll(); }

    public int getAvailableSeatsForTrip(Trip trip) { return repo.getAvailableSeatsForTrip(trip); }
}

File: src/service/ReservationService.cs
using System;
using System.Collections.Generic;
using AgentiiDeTurism.src.domain;
using AgentiiDeTurism.src.repository;

namespace AgentiiDeTurism.src.service
{
    public class ReservationService
    {
        private IReservationRepo repo;

        public ReservationService(IReservationRepo repo)
        {
            this.repo = repo;
        }

        public void saveReservation(String client, Trip trip, String phoneNumber, int seats)
        {
            repo.save(new Reservation(new Tuple<string, Trip>(client, trip), phoneNumber, seats));
        }

        public ICollection<Reservation> getAllReservations() { return repo.getAll(); }

        public int getAvailableSeatsForTrip(Trip trip) { return repo.getAvailableSeatsForTrip(trip); }
    }
}
